src/geometry.F90

```diff
@@ -1582,5 +1614,327 @@ subroutine handle_lost_particle(p, message)
+
+      ! ====================================================================
+      ! CELL CONTAINS LOWER UNIVERSE, RECURSIVELY FIND CELL
+      elseif (c % type == CELL_FILL) then
+        ! Set offset for the cell on this level
+        c % offset(map) = offset
+
+        ! Count contents of this cell
+        next_univ => universes(c % fill)
+        offset = offset + count_target(next_univ, kounts, found, goal, map)
+
+        ! Move into the next universe
+        next_univ => universes(c % fill)
+        c => cells(cell_index)
+
+      ! ====================================================================
+      ! CELL CONTAINS LATTICE, RECURSIVELY FIND CELL
+      elseif (c % type == CELL_LATTICE) then
+
+        ! Set current lattice
+        lat => lattices(c % fill) % obj
+
+        select type (lat)
+
+        type is (RectLattice)
+        
+          ! Loop over lattice coordinates
+          do j = 1, lat % n_cells(1)
+            do k = 1, lat % n_cells(2)
+              do m = 1, lat % n_cells(3)
+                lat % offset(map, j, k, m) = offset
+                next_univ => universes(lat % universes(j, k, m))
+                offset = offset+count_target(next_univ,kounts,found,goal,map)
+              end do
+            end do
+          end do
+
+        type is (HexLattice)
+
+          ! Loop over lattice coordinates
+          do m = 1, lat % n_axial
+            do k = 1, 2*lat % n_rings - 1
+              do j = 1, 2*lat % n_rings - 1
+                ! This array location is never used
+                if (j + k < lat % n_rings + 1) then
+                  cycle
+                ! This array location is never used
+                else if (j + k > 3*lat % n_rings - 1) then
+                  cycle
+                else
+                  lat % offset(map, j, k, m) = offset
+                  next_univ => universes(lat % universes(j, k, m))
+                  offset = offset+count_target(next_univ,kounts,found,goal,map)
+                end if
+              end do
+            end do
+          end do
+        end select
+
+      end if
+    end do
+    
+    return
+        
+  end subroutine calc_offsets
+  
+!===============================================================================
+! COUNT_TARGET recursively totals the numbers of occurances of a given
+! universe ID beginning with the universe given.
+!===============================================================================
+
+  recursive function count_target(univ, kounts, found, goal, map) result(kount)
+
+    type(Universe), intent(inout) :: univ         ! universe to search through
+    integer, intent(inout)        :: kounts(:,:)  ! Target count
+    logical, intent(inout)        :: found(:,:)   ! Target found
+    integer, intent(in) :: goal            ! target universe ID
+    integer             :: kount           ! number of times target located
+    integer             :: map             ! current map
```

Comment:
Good point, just fixed this and cleaned up the variable declarations for ``calc_offsets`` and ``count_target``